salt/utils/decorators/extension_deprecation.py

```diff
@@ -0,0 +1,35 @@
+"""
+Decorators for deprecation of modules to Salt extensions
+"""
+import logging
+from functools import wraps
+
+import salt.utils.args
+import salt.utils.versions
+
+log = logging.getLogger(__name__)
+
+
+def extension_deprecation_message(
+    version_number, version_name, extension_name, extension_repo
```

Comment:
```suggestion
    version, extension_name, extension_repo
```
Just pass version and parse it through `salt.version.SaltStackVersion`?
That way you don't have to pass `3009` **and** `Potassium`?
From the name we can get the version and from the version we can get the name if we have a SaltStackVersion...

https://github.com/saltstack/salt/blob/master/salt/utils/versions.py#L141-L161

Reply:
Actually, just pass the name, use it in the message, and `warn_until` will turn that into the right type on the first argument.

Reply:
FWIW, I think Salt version names (e.g. Potassium) are a pointless layer of abstraction now that we have predictable version numbers. `3009` means something to me, but I'd have to look up the translation of `Potassium` to enter it for this message, only to have it converted back for `warn_until`.

Reply:
> FWIW, I think Salt version names (e.g. Potassium) are a pointless layer of abstraction now that we have predictable version numbers. `3009` means something to me, but I'd have to look up the translation of `Potassium` to enter it for this message, only to have it converted back for `warn_until`.

Agreed. Passing/using the version number, instead of the name makes it clearer(now that version numbering is predictable)